test/functional/terminal/highlight_spec.lua

```diff
@@ -248,8 +248,9 @@ it('CursorLine and CursorColumn work in :terminal buffer in Normal mode', functi
     {1:bar fooba^r                                        }|
                                                       |
   ]])
-  -- CursorLine and CursorColumn are combined with TermCursorNC.
-  command('highlight TermCursorNC gui=reverse')
+  -- CursorLine and CursorColumn are combined with other highlight groups
+  command('highlight Search NONE gui=reverse')
+  command('match Search /\\%6l\\%11c/')
```

Comment:
Match highlight does not come from the terminal and does not cover the drawline.c terminal highlight code. The highlights need to come from the program running in the terminal.

Reply:
See https://github.com/neovim/neovim/pull/31562#discussion_r1884787479

Reply:
?

Reply:
TermCursorNC doesn’t come from the program running in the terminal either? What is this supposed to be testing? Your PR adding this test had zero context or other information.

By the way, editing your comment after the fact and then asking if I didn’t understand it is poor form.

Reply:
> By the way, editing your comment after the fact and then asking if I didn’t understand it is poor form.

After what fact? I only made a one-character edit to my comment before I saw your reply, and it didn't change the overall meaning.

> By “comes from the terminal” do you mean a highlight that has the `kHlTerminal` kind?

No, I mean it needs to come from `terminal_get_line_attributes()`.

Reply:
>No, I mean it needs to come from terminal_get_line_attributes(). But since TermCursorNC is removed, it can now only come from the program running in the terminal.

Ok the updated test ought to be sufficient now I think. Instead of the old cursor I printed text to the terminal using different SGRs.